WcaOnRails/app/views/competitions/index.html.erb

```diff
@@ -1,17 +1,125 @@
-      </tr>
+  <h2><%= yield(:title) %></h2>
+
+  <%= form_tag competitions_path, method: :get, class: "competition-select form-inline" do %>
+    <% if @competitions.length == 0 %>
+      <%= render 'index_form' %>
+      <div class="alert alert-warning">No competitions found!</div>
+    <% else %>
+      <% if params[:display] == "List" %>
+        <table class="table competitions-table table-striped table-hover table-condensed floatThead">
+          <thead>
+            <tr>
+              <th colspan="7">
+                <%= render 'index_form' %>
+              </th>
+            </tr>
+            <tr>
+              <th>Year</th>
+              <th>Date</th>
+              <th>Name</th>
+              <th>Country, City</th>
+              <th>Venue</th>
+              <th class="big-column"></th>
+            </tr>
+          </thead>
+          <tbody>
+            <%= render 'index_table', competitions: @not_past_competitions %>
+            <tr id="today-line">
+              <td class="text-center"><a id="today"></a>Today</td>
+              <td colspan="6" ></td>
+            </tr>
+            <%= render 'index_table', competitions: @past_competitions %>
+          </tbody>
+        </table>
+      <% else %>
+        <%= render 'index_form' %>
+        <div id="map"></div>
+      <% end %>
     <% end %>
-  </table>
+  <% end %>
 </div>
+
+<script>
+function initMap() {
+  var $map = new google.maps.Map(document.getElementById('map'), {
+    zoom: 2,
+    center: {lat: 0, lng: 0},
+    scrollwheel: true
+  });
+
+  allCompetitionsMap(<%= @competitions.map do |c|
+  {
+    id: c.id,
+    name: c.name,
+    latitude_degrees: c.latitude_degrees,
+    longitude_degrees: c.longitude_degrees,
+    cityName: c.cityName,
+    marker_date: c.start_date.to_formatted_s(:long),
+    is_over: c.is_over?,
+    url: competition_path(c)
+  }
+    end.to_json.html_safe %>, $map);
+}
+
+function allCompetitionsMap(competitions, map) {
+
+  var markers = [];
+
+  competitions.forEach(function(c) {
+
+    var contentString = "<a href=" + c.url + ">" + c.name + "</a><br />" + c.marker_date + " - " + c.cityName;
+
+    var infowindow = new google.maps.InfoWindow({
+      content: contentString
+    });
+
+    if (c.is_over) {
+      iconImage = 'http://maps.google.com/mapfiles/ms/icons/blue.png';
+    }
+    else {
+      iconImage = 'http://maps.google.com/mapfiles/ms/icons/red.png';
+    }
+
+    c.marker = new google.maps.Marker({
+      map: map,
+      position: {
+        lat: c.latitude_degrees,
+        lng: c.longitude_degrees,
+      },
+      title: c.name,
+      icon: iconImage
+    });
+
+    c.marker.addListener('click', function() {
+      infowindow.open(map, c.marker);
+    });
+
+    markers.push(c.marker);
+  });
+
+  var markerCluster = new MarkerClusterer(map, markers, {
+    maxZoom: 10,
+    clusterSize: 30
+  });
+}
+
+function scrollToCenter() {
```

Comment:
Oh, I meant to do something generic that other parts of our codebase could use, like what ccm did. Can this get pulled out to application.js?